### ISSUE TITLE
Release tests failing on DockerYmlTestSuiteIT

### ISSUE THREAD

**Issue by hub-cap** [elastic/elasticsearch#54513]
[build 1060](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/1060/) -- [build scan 1060](https://gradle-enterprise.elastic.co/s/r2fkflx3sr4rm)
[build 1059](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/1059/) -- [build scan 1059](https://gradle-enterprise.elastic.co/s/jlhouiqj73l44)
[build 1058](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/1058/) -- [build scan 1058](https://gradle-enterprise.elastic.co/s/ceef2c3bqhw5a)
[build 1057](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/1057/) -- [build scan 1057](https://gradle-enterprise.elastic.co/s/dqnwuotarg5uy)

```
DockerYmlTestSuiteIT » test {p0=/11_nodes/Test cat nodes output} | 3.863s |  
-- | -- | --
DockerYmlTestSuiteIT » test {p0=/10_info/Info} | 0.409s |  
DockerYmlTestSuiteIT » test {p0=/11_nodes/Additional disk information} | 0.407s |  
DockerYmlTestSuiteIT » test {p0=/11_nodes/Test cat nodes output with full_id set} | 0.390s |  
```

**Comment by gwbrown** [comment]
`{"error":"no handler found for uri [_index_template/*] and method [DELETE]"}`

I don't think we have any (released) APIs under `_index_template` so I did a quick check, looks like this might be related to index templates v2 - this is the only place I was able to find a reference to `_index_templates/*` with the DELETE verb:
https://github.com/elastic/elasticsearch/blob/5a98fc20e106c4f2acc1411c66bd8e796a8051d8/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L552-L567

**Comment by mark-vieira** [comment]
@danhermann It looks like we are now running into a similar error due to the merge of #54728. Should we do something similar to #54517 and be more lenient when cleaning up data streams in the rest tests as well?

https://gradle-enterprise.elastic.co/s/shkpegiom4rw2/tests/gkkty2gnquabk-4juckqsqlr3fc?open-stacktraces=WzBd

**Comment by danhermann** [comment]
@mark-vieira, that PR was temporarily commented out in master, so I think the same should happen in `7.x`. I'll put a PR up shortly.

**Comment by danhermann** [comment]
Actually, I think https://github.com/elastic/elasticsearch/pull/54959 is what you need here?

**Comment by mark-vieira** [comment]
Yep, thanks @danhermann. Looks like that should clear things up as well.

**Comment by rjernst** [comment]
fixed by #54959
